### ISSUE TITLE
[Bug]: Terraform 1.5.0-rc1 not generating config file after import and plan as mentioned in the release doc.

### ISSUE THREAD

**Issue by dev-travelex** [hashicorp/terraform-provider-aws#31728]
### Terraform Core Version

1.5.0-rc1 on windows_amd64

### AWS Provider Version

v5.1.0

### Affected Resource(s)

As per the documentation given here : https://github.com/hashicorp/terraform/releases

`Generating configuration for imported resources: in conjunction with the import block, this feature enables easy templating of configuration when importing existing resources into Terraform. A new flag -generate-config-out=PATH is added to terraform plan. When this flag is set, Terraform will generate HCL configuration for any resource included in an import block that does not already have associated configuration, and write it to a new file at PATH. Before applying, review the generated configuration and edit it as necessary.`

```

PS C:\Users\devang.sanghan\Projects\terraform-import> terraform plan -generate-config-out=terraform-import 
data.aws_caller_identity.this: Reading...
aws_instance.ansible-ec2: Refreshing state... [id=i-0444b96xxxxxx]
data.aws_caller_identity.this: Read complete after 1s [id=3548xxxxx]


Terraform used the selected providers to generate the following execution plan. Resource actions
with the following symbols:
  - destroy

Terraform will perform the following actions:

   aws_instance.ansible-ec2 will be destroyed
   (because aws_instance.ansible-ec2 is not in configuration)

```  

### Expected Behavior

I expect the config file to be created based on the import done and then the plan run.

### Actual Behavior

Terraform proposes destroying the resources as it doesn't find those resources in the config file.

### Terraform Configuration Files

No config created as it is expected for this terraform version to create them based on the import and the plan.

### Steps to Reproduce

Use terraform 1.5.0-rc1 and then run `terraform import` on an existing ec2 instance created manually. 
After successful import, run `terraform plan -generate-config-out={PATH for config file}`.

### Debug Output

N/A

### Panic Output

N/A

### Important Factoids

N/A

### References

N/A

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @dev-travelex 👋  I'm happy to follow up with the Terraform Core team if I'm misunderstood, however, I've got a different read on how the `-generate-config-out=$PATH` functionality works, specifically this part:

> When this flag is set, Terraform will generate HCL configuration for any resource included in an `import` block that does not already have associated configuration, and write it to a new file at `PATH`.

My read on this is that it isn't meant to generate configuration after running `terraform import` to import resources, but is instead related to the new `import` blocks that are described in the same CHANGELOG entry.

**Comment by dev-travelex** [comment]
Hi @justinretzolk , Could you please elaborate on what exactly does it do then?

**Comment by justinretzolk** [comment]
@dev-travelex This question is probably better served by the Core team, and will likely be addressed more in-depth when Terraform 1.5.0 is released, however, my takeaway based on the changelog is that once you define an `import` block within the configuration to allow Terraform to import an existing resource, the `-generate-config-out=$PATH` argument can then be used to generate the necessary HCL configuration that goes along with the resource that's being imported (note that this is not referring to resources that are imported with the existing `terraform import` command).

**Comment by dev-travelex** [comment]
I see the documentation for v1.5 now and it is to be done differently from how I had assumed it. So, closing this issue. Thanks!
